### PR TITLE
27 u bar controller

### LINKED ISSUE #27: U bar Controller
Controls the upper arm of the robot.

Finished when: The extension is fully functional.

### PR DESCRIPTION
**Pull request by mistrspidr**
Code for Screw, U Bar, and Intake

⚠HAS NOT BEEN TESTED⚠

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -10,7 +10,7 @@ buildscript {
         google()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:7.2.2'
+        classpath 'com.android.tools.build:gradle:7.3.0'
```
Comment (Carl4) [inline build.gradle:13]: Not sure if we should all upgrade to gradle 7.3.0

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/util/TowerController.java [hunk 2]
```diff
@@ -1,32 +1,54 @@
...
         screw = hardwareMap.get(DcMotor.class, "screw");
+        uBar = hardwareMap.get(DcMotor.class, "uBar");
+        intake = hardwareMap.get(Servo.class, "intake");
         screw.setDirection(DcMotor.Direction.FORWARD);
+        uBar.setDirection(DcMotor.Direction.FORWARD);
 
         //Setup sensors
         DigitalChannel highSensor = hardwareMap.get(DigitalChannel.class, "highSensor");
         DigitalChannel lowSensor = hardwareMap.get(DigitalChannel.class, "lowSensor");
+        DigitalChannel uBarSensor = hardwareMap.get(DigitalChannel.class, "uBarSensor");
```
Comment (Carl4) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/util/TowerController.java]: What's uBarSensor supposed to be? A digital input?

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/util/TowerController.java [hunk 3]
```diff
@@ -45,5 +67,35 @@ public void handleScrew() {
...
+        if (uBarNum == 3) {
+            drive(COUNTS_PER_INCH * 60, 1);
+        }
+        if (uBarNum == 4) {
+            drive(COUNTS_PER_INCH * 80, 1);
+        }
+    }
+
+    private void drive(double uBarTarget, double speed) {
+        uBarLevel += uBarTarget;
```
Comment (Carl4) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/util/TowerController.java]: This looks like you're adding the target to the current bar level. But I think you want to just set the target to the new value.

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/util/TowerController.java [hunk 4]
```diff
@@ -45,5 +67,35 @@ public void handleScrew() {
...
+    public void handleIntake (HardwareMap hardwareMap) {
+        if(intakePos){
+            intake.setPosition(1);
+        }
+        else{
+            intake.setPosition(0);
+        }
+    }
 }
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        //hi. you found me. -SECRET COMMENT
```
Comment (Carl4) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/util/TowerController.java:97]: I think you should make a handleTower() function that handles all the components of the tower: handleIntake, handleBar, handleScrew.

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/util/TowerController.java (hunk 3 above)
Comment (mistrspidr) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/util/TowerController.java]: I tried removing it and it came up with an error. I don't know why, but the plus sign is definitely necessary.

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/util/TowerController.java (hunk 4 above)
Comment (mistrspidr) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/util/TowerController.java:97]: No, that's for our main code for TeleOp

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/util/TowerController.java (hunk 2 above)
Comment (mistrspidr) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/util/TowerController.java]: It was never used so I removed it.
No Errors

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/util/TowerController.java (hunk 3 above)
Comment (Carl4) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/util/TowerController.java]: I see the issue.  `uBarLevel` is an integer but `uBarTarget` is a double.  Try making `uBarTarrget` an integer and you should be able to make it `uBarLevel = uBarTarget`

**Pull request merged**